docs/development/dep-lib-extension-hooks.md

```diff
@@ -0,0 +1,154 @@
+The `AbortReconcile` values of all hook results from hooks with this type will be ANDed instead of ORed before being evaluated.
+
+**BeforeAnyReconcile** hooks are executed before any of the mentioned `Abort`, `Force-Reconcile`, `Delete`, or `Reconcile` steps, while **BeforeAbort**, **BeforeForceReconcile**, **BeforeDelete**, and **BeforeReconcile** are only executed before their respective step (meaning only one of these hook types is executed each time). While it is possible to abort the reconciliation by setting the hook result's `AbortReconcile` to `true`, this will have the same consequences as mentioned above and should only be done with extreme caution.
+
+Hooks with type **End** are executed at the end of the reconciliation, after the deployer-specific code has returned.
+
+
+## Registering Hooks and Implementing the Interface
+
+To enable the extension hooks, an additional method has been added to the `Deployer` interface.
+```golang
+// ExtensionHooks returns all registered extension hooks.
+ExtensionHooks() extension.ReconcileExtensionHooks
+```
+
+This method has to be implemented so that it returns all registered hooks as a mapping from hook types to lists of hook functions.
+```golang
+// ReconcileExtensionHooks maps hook types to a list of hook functions.
+type ReconcileExtensionHooks map[HookType][]ReconcileExtensionHook
+```
+
+The easiest way of implementing this is just adding said map to the deployer struct and returning it in the `ExtensionHooks` method, see e.g. the mock deployer for an example:
+```golang
+type deployer struct {
+	log        logr.Logger
+	lsClient   client.Client
+	hostClient client.Client
+	config     mockv1alpha1.Configuration
+	hooks      extension.ReconcileExtensionHooks
+}
+
+func (d *deployer) ExtensionHooks() extension.ReconcileExtensionHooks {
+	return d.hooks
+}
+```
+
+The extension library offers a helper method to register a new hook:
+```golang
+// RegisterHook appends the given hook function to the list of hook functions for all given hook types.
+// It returns the ReconcileExtensionHooks object it is called on for chaining.
+func (hooks ReconcileExtensionHooks) RegisterHook(hook ReconcileExtensionHook, hypes ...HookType) ReconcileExtensionHooks
```

Comment:
This was intended, but let's discuss this: I used `hype` for the singular, as `type` is a reserved keyword in go (see below). Should I use `types` for the plural as this can be used or should I stick to the `hype` scheme and call it `hypes`?

Reply:
See alos my answer below.
I think this is a matter of taste, and one should not argue about tastes ;-)
However, `hypes`or `hype` always looks like a typo for me.
I would rather go for an abbreviation like `ht`.
Final decision is up to you.